### ISSUE TITLE
[Lev] Apple FM decision-model roadmap and tracking

### ISSUE THREAD

**Issue by AtlantisPleb** [OpenAgentsInc/openagents#9345]
## Summary

Build a third implementation of the System One contract, answered by Apple's on-device foundation model, ending at a `lev-serve` binary that a `crates/jev` client reaches with a `base_url` change and a calibration record behind every probability it reports.

Jev (`crates/jev`, hosted), Kev (`crates/kev`, open weights, #9337), and Lev (`crates/lev`, Apple on-device) all speak the same `POST /v1/systemone` contract. A caller picks by `base_url`.

Background: `docs/lev/` — `architecture.md` for the design and why kev's mechanism does not port, `apple-fm-surface.md` for what this workspace already established, `calibration.md` for the measurement rule, `mesh-plan.md` for the fleet path, `roadmap.md` for this sequence.

## Why kev's mechanism does not port

Apple's framework returns text and typed structured values. It exposes no logits, no log-probabilities, and no hidden states, and it runs no code of ours inside the model. Kev's pointer head has no position to read and no head to run. Lev derives a distribution from observable behavior and then earns the word "probability" by fitting a calibration map on labelled outcomes.

## Sequence

1. Behavior measurement harness and the record — [x] met in `docs/lev/measurements/2026-09-19-behavior.md` (#9346)
2. `crates/lev`: contract types, renderer, schema compiler, typed refusals — [x] met in `crates/lev/src` (#9347)
3. The bridge seam and question isolation — [x] met in `crates/lev/src/bridge.rs`, admission floor in 5ededb8e0 (#9348, #9389 live probe recorded in `dbab6bef920ed77e45cb2fd74f1245ab90e44ec6` and `04ef00cb5e`)
4. Estimators: seeded ensemble and argmax — [x] met in `crates/lev/src/estimator.rs` (#9349)
5. Suites and the calibration map — [x] met with a caveat: `routing` admitted on `support-v2`, `severity` and `urgency` refused (#9350, `docs/lev/measurements/2026-09-19-adapter-v1.md`)
6. `lev-serve`: the door and its refusals — [x] met in `crates/lev/src/bin/lev_serve.rs` (#9351)
7. The band readout — [x] met with a caveat: constant on the base, monotone on the band adapter (#9352, `2026-09-19-adapter-band.md`)
8. The adapter lane (deferred: Apple's adapter training toolkit is an external dependency) — [x] met in `crates/lev/src/adapter.rs`, `training/lev-adapter/` (#9353); training run is #9363
9. Comparison and disposition — [x] met in `docs/lev/disposition.md` (#9354)
10. Mesh row (gated on a licensing read) — [x] closed, licensing yes (#9355); the behavioral floor is #9389

## Rules

- Lev is not Jev and it is not kev. No probability leaves the door without a fitted calibration map recorded against the serving base signature.
- The model proposes, the host acts and reports. A typed answer is a proposal, never evidence that anything happened.
- The state is data, never instruction. Policy goes in session instructions; caller state goes in the prompt, marked as data.
- Everything lands in this repository. No dependency on a sibling repository, an external checkout, or a vendor toolkit.
- Rust only, no TypeScript, no GitHub-billed automation. The Swift helper is in-repo, signed, and its signature is checked before launch.
- Every issue lands with its tests and its docs update.

## Follow-on issues, reconciled against `main` on 2026-09-20

- [x] #9363 first adapter — all four acceptance numbers published: `docs/lev/measurements/2026-09-19-adapter-v1.md`, `2026-09-19-adapter-band.md`, `2026-09-19-flip-rate-variance.md`
- [x] #9389 behavioral admission floor — met in `dbab6bef920ed77e45cb2fd74f1245ab90e44ec6` and `04ef00cb5e`: real Apple runtime observations, admitted/refused result, wall time, and retained 25-call choice capture; `docs/lev/measurements/2026-09-20-admission-live.md`
- [x] #9380 adapters out of domain — met in `0a6b30b76d`: complete band/permutation coding passes, base external pass, full probability panels, disposition, and immutable observation references; `docs/decision-models/2026-09-20-lev-domain-gap.md`

**Comment by AtlantisPleb** [comment]
Sequence status after the first working session.

**Done and on `main`:** #9346 behavior record, #9347 contract types and schema compiler, #9348 bridge seam and isolation, #9349 estimators, #9351 `lev-serve`.
**Blocked on a measurement:** #9350 and #9352 — no raw signal varies with correctness; see the comments there.
**Deferred:** #9353, Apple's toolkit is an external dependency.
**Open:** #9354 — a first run exists in `docs/lev/measurements/2026-09-19-comparison.md`; the written disposition is still owed.
**Closed:** #9355 — licensing resolved yes.

One client, three doors, live:

    jev (hosted)   8 of 8   247 ms
    kev            1 of 8   182 ms   (see #9342)
    lev            7 of 8 2,082 ms

Two findings worth carrying out of this lane: Apple's guardrails refuse a delimiter-fenced state regardless of its content, so the standard prompt-injection defense is unavailable on that runtime; and an N-draw ensemble is N chances to trip a guardrail, so the estimator tolerates a minority of refused draws.

**Comment by AtlantisPleb** [comment]
## Sequence complete

Every issue is closed except #9353, the adapter lane, which is deferred on an external toolkit.

| # | Issue | Outcome |
| --- | --- | --- |
| 1 | #9346 behavior record | done — greedy deterministic, seeds reproduce 16/16 in and across processes |
| 2 | #9347 contract types, schema compiler | done — 37 tests |
| 3 | #9348 bridge seam, isolation | done — sibling 0.62, absent 0.62, state 1.00 live |
| 4 | #9349 estimators | done — L1, L2, L3, plus a helper pool |
| 5 | #9350 calibration map | done, negative — machinery works, no map passes the gate at this suite size |
| 6 | #9351 `lev-serve` | done — an unmodified `jev` client round-trips all three types and lists models |
| 7 | #9352 band readout | done, negative — the band is constant on the base model |
| 8 | #9353 adapter lane | deferred — Apple's toolkit is an external dependency |
| 9 | #9354 comparison and disposition | done — `docs/lev/disposition.md` |
| 10 | #9355 mesh row | closed — licensing resolved yes |

## Where Lev stands against Jev and Kev

| Door | Accuracy | ECE | Brier | Latency |
| --- | --- | --- | --- | --- |
| jev (hosted) | 0.96 | 0.099 | 0.026 | ~220 ms |
| kev-0.5b | 0.88 | 0.176 | 0.122 | ~180 ms |
| kev-4b | 0.77 | 0.188 | 0.149 | ~1 s |
| lev | 0.85 | 0.087 | 0.127 | ~1,600 ms |

## How to improve it, in order

1. ~~Sample concurrently~~ — **built**. `bridge::Pool` plus `l2_pool`; 2,365 ms to 1,560 ms on eight samples. Only 1.5x from four helpers, because the on-device model is one shared resource and the callers queue for it. **Concurrency is capped by the device, not the bridge**, so this lever is now spent.
2. **Serve L1 by default.** Greedy is deterministic and costs one call at ~300 ms. Now the largest remaining latency lever.
3. **Grow the suite to ~150–200 items per family.** The only thing between the existing calibration machinery and an admitted map. Authoring work, not research.
4. **Average over option permutations.** Order flips one greedy answer in eight, against kev-0.5b's 7.4% and Jev's zero.
5. **Train an adapter (#9353).** The only route to the accuracy gap and to a band signal.
6. **Do not chase Jev on sharpness.** Every Lev number is counted from samples at resolution 1/N; matching Brier 0.026 costs the latency that makes Lev worth having.

## Two findings worth carrying out of this lane

- **Apple's guardrails refuse a delimiter-fenced state regardless of content.** `<state>` tags, bare tags, and triple quotes all refused on the same benign item; a plain label answered. The standard prompt-injection defense is unavailable on that runtime.
- **An N-draw ensemble is N chances to trip a guardrail.** The estimator tolerates a minority of refused draws and refuses only when every draw refuses.

**Comment by AtlantisPleb** [comment]
## Adapter lane: environment complete (#9353 closed), training filed as #9363

Every Lev issue is now closed except the tracker and the training run, which is blocked on an Apple Developer download rather than on work.

### What the adapter work established

**Apple's runtime loads a `.fmadapter` package this repository wrote.** `crates/lev/src/adapter.rs` reimplements the format from its published shape — Core ML blob storage with a 64-byte header, aligned records, `0xdeadbeef` magic, fp16 payloads, plus the camelCase metadata rules — and `tests/adapter.rs` hands one to the device, which loads it and round-trips its producer metadata. A companion test asserts that *generating* with it fails, because its tensors are junk; that assertion is the tripwire for when real weights arrive.

**The live base signature is readable from outside the framework.** `compatibleAdapterIdentifiers(name:)` returns `fmadapter-lev-9799725`, which is how you learn which toolkit version to download. It still matches what was documented in March 2026, so the base has not moved on macOS 26.4 (25E246).

**Train and serve share one renderer, and it is proved rather than assumed.** The harness is Python because Apple's toolkit is, so there are two implementations of one set of rules; `check_parity.py` fails if they disagree on a character. Currently 52 of 52.

### Why the toolkit is not in the repo

Apple distributes it to Developer Program members who accept its terms, which do not permit redistribution, and this repository is public. `training/lev-adapter/toolkit/` is gitignored, and the download step is in `NEEDS_OWNER.md` at the workspace root with the signature prefix to match.

### Non-Rust trees, and the reason for each

`AGENTS.md` says Rust-only. There are now two exceptions, each for a reason rather than a convenience: `swift/lev-bridge` because `FoundationModels` has no Rust binding, and `training/lev-adapter` because Apple's toolkit is Python and there is no other way to produce a `.fmadapter`. Both produce artifacts Rust consumes, and both contracts are tested.

### The standing risk

An adapter is pinned to a base signature that ships with the operating system. An OS update invalidates the adapter and every calibration map fitted with it. `lev-serve` refuses to start on a mismatch rather than serving a bad pairing.

**Comment by AtlantisPleb** [comment]
Reconciled the remaining stale tracker entries against published evidence:

- Sequence item 3 now links the completed live admission record instead of saying the probe is owed.
- #9389 is checked: `dbab6bef920ed77e45cb2fd74f1245ab90e44ec6` and `04ef00cb5e` retain the real runtime result and supplemental 25-call observation.
- #9380 is checked: `0a6b30b76d` retains the complete out-of-domain runs, probability panels, disposition, and immutable references.

All ten sequence entries and all listed follow-ons now read met, and all thirteen referenced child issues are closed. Closing this Lev roadmap tracker. This does not close the separate #9382/#9393 latency work or #9426 serving-verification failure; those remain open.
